Emby.Server.Implementations/TV/TVSeriesManager.cs

```diff
@@ -140,7 +140,15 @@ public IEnumerable<Episode> GetNextUpEpisodes(NextUpQuery request, User user, IE
             var currentUser = user;
 
             var allNextUp = seriesKeys
-                .Select(i => GetNextUp(i, currentUser, dtoOptions, request.Rewatching));
+                .Select(i => GetNextUp(i, currentUser, dtoOptions, false));
+
+            if (request.EnableRewatching)
+            {
+                allNextUp = allNextUp.Concat(
+                    seriesKeys.Select(i => GetNextUp(i, currentUser, dtoOptions, true))
```

Comment:
Double enumeration of serieskeys

Reply:
Do you have any recommendations on how to avoid it? The first pass gets the next unplayed episode and the second pass gets the next played episode based on the most recently watched episode. I could not think of a better approach to handle that. :sweat_smile:

Reply:
Change the signature to `IReadOnlyList<string> seriesKeys`